### PR TITLE
Fix/schema validate

### PR DESCRIPTION
**Pull request by vikaskumr**

### CODE REVIEW

File: apps/dsb-client-gateway-api/src/app/modules/utils/validator/decorators/IsSchemaValid.ts [hunk 1]
```diff
@@ -19,25 +47,28 @@ export function IsSchemaValid(
...
+    );
+  }
 
   try {
-    const jsonPayload = JSON.parse(payload);
     validate = ajv.compile(schema);
-    valid = validate(jsonPayload);
   } catch (e) {
-    valid = false;
+    console.log(e);
```
Comment (hejkerooo) [inline apps/dsb-client-gateway-api/src/app/modules/utils/validator/decorators/IsSchemaValid.ts]: remove/replace with logger

File: apps/dsb-client-gateway-api/src/app/modules/utils/validator/decorators/IsSchemaValid.ts [hunk 2]
```diff
@@ -19,25 +47,27 @@ export function IsSchemaValid(
 
 function validateJSONSchema(schema: object, payload: string) {
   let validate: any;
-  let valid: boolean;
+  let jsonPayload: object;
+  try {
+    jsonPayload = JSON.parse(payload);
+  } catch (e) {
+    throw new SchemaNotValidException(
```
Comment (hejkerooo) [inline apps/dsb-client-gateway-api/src/app/modules/utils/validator/decorators/IsSchemaValid.ts:54]: make another exception for it - malformed JSON or something like this.

File: apps/dsb-client-gateway-api/src/app/modules/message/service/message.service.ts [hunk 3]
```diff
@@ -101,6 +102,14 @@ export class MessageService {
       throw new TopicNotFoundException('NOT Found');
     }
 
+    const topicFound = channel.conditions.topics.find(
+      (topicOfChannel) => topicOfChannel.topicId === topic.id
+    );
+
+    if (!topicFound) {
+      throw new TopicNotRelatedToChannelException('');
```
Comment (hejkerooo) [inline apps/dsb-client-gateway-api/src/app/modules/message/service/message.service.ts:110]: why empty message? pass there topic name / owner so user is informed about it

File: apps/dsb-client-gateway-api/src/app/modules/message/service/message.service.ts [hunk 4]
```diff
@@ -101,6 +102,14 @@ export class MessageService {
       throw new TopicNotFoundException('NOT Found');
     }
 
+    const topicFound = channel.conditions.topics.find(
```
Comment (hejkerooo) [inline apps/dsb-client-gateway-api/src/app/modules/message/service/message.service.ts:105]: this method is already big - can we move this validation to separate method i.e. `checkTopicForChannel` or something like that?

File: apps/dsb-client-gateway-api/src/app/modules/utils/validator/decorators/IsSchemaValid.ts (hunk 1 above)
Comment (vikaskumr) [reply apps/dsb-client-gateway-api/src/app/modules/utils/validator/decorators/IsSchemaValid.ts]: this is already removed

File: apps/dsb-client-gateway-api/src/app/modules/message/service/message.service.ts [hunk 5]
```diff
@@ -101,6 +117,17 @@ export class MessageService {
...
     }
 
+    const isTopicRelatedToChannel: boolean = this.checkTopicForChannel(
+      channel,
+      topic
+    );
+
+    if (!isTopicRelatedToChannel) {
+      throw new TopicNotRelatedToChannelException(
+        `topic with ${topic.name} not related to channel with name ${dto.fqcn}`
```
Comment (hejkerooo) [inline apps/dsb-client-gateway-api/src/app/modules/message/service/message.service.ts:127]: add topic owner

File: apps/dsb-client-gateway-api/src/app/modules/message/service/message.service.ts [hunk 6]
```diff
@@ -82,10 +83,25 @@ export class MessageService {
...
   }
 
+  private checkTopicForChannel(
+    channel: ChannelEntity,
+    topic: TopicEntity
+  ): boolean {
+    const topicFound = channel.conditions.topics.find(
+      (topicOfChannel) => topicOfChannel.topicId === topic.id
+    );
+    if (!topicFound) {
```
Comment (hejkerooo) [inline apps/dsb-client-gateway-api/src/app/modules/message/service/message.service.ts:93]:
```
return !channel.conditions.topics.find(
      (topicOfChannel) => topicOfChannel.topicId === topic.id
    );
```

**Review by hejkerooo — APPROVED** [review]

**Review by hejkerooo — APPROVED** [review]

**Pull request merged**